### ISSUE TITLE
PCC drop in Gemma3(text extraction and translation), PCC=0.8284904120505975

### ISSUE THREAD

**Issue by kamalrajkannan78** [tenstorrent/tt-xla#4608]
### Describe the bug

- `gemma/text_extraction_and_translation/pytorch-Translategemma_4B_IT-single_device-inference` fails with `PCC=0.8284904120505975` - [may11_gemma_4b_it_1.log.zip](https://github.com/user-attachments/files/27605265/may11_gemma_4b_it_1.log.zip)

### Root cause analysis

- Since logits themselves already have a low PCC (0.87351), the focus should be on identifying the earliest point where PCC drops below 0.99.
- The vision part(SiglipVisionModel) contains 27 encoder blocks.
- Below is the observed relationship between the number of encoder blocks and PCC:

| Iteration | PCC                |
|-----------|--------------------|
| 0         | 0.9999194640612024 |
| 13        | 0.9977553226336904 |
| 15        | 0.9933323153917724 |
| 16        | 0.9881700005064309 |
| 17        | 0.9789576338014355 |
| 20        | 0.9726276817410779 |

- As the number of encoder blocks increases, PCC gradually decreases, dropping below 0.99 on 16th iteration - [exps.zip](https://github.com/user-attachments/files/27605390/exps.zip)
- This indicates the accumulated PCC degradation between many layers
-  It requires robust tooling to debug.
- Note : On FP32, model is having OOM - [may12_gemma_4b_it_fp32.log.zip](https://github.com/user-attachments/files/27622580/may12_gemma_4b_it_fp32.log.zip), [loader.py](https://github.com/user-attachments/files/27622566/loader.py)

### Repro

```
git checkout may11_gemma3_pcc_drop
git submodule update --init --recursive

pytest tests/runner/test_models.py::test_all_models_torch[gemma/text_extraction_and_translation/pytorch-Translategemma_4B_IT-single_device-inference] -svv

# to runs experiments use the attached version of code and re-run above commands
```
